vcd/resource_vcd_nsxv_distributed_firewall.go

```diff
@@ -0,0 +1,647 @@
+		excludeSource := false
+		if rule.Sources != nil && len(rule.Sources.Source) > 0 {
+			var rawSourceList []interface{}
+			if rule.Sources.Excluded {
+				excludeSource = true
+			}
+			for _, s := range rule.Sources.Source {
+				sourceMap := map[string]interface{}{
+					"name":  s.Name,
+					"type":  s.Type,
+					"value": filterVdcId(s.Value),
+				}
+				rawSourceList = append(rawSourceList, sourceMap)
+			}
+			sourceSet := schema.NewSet(schema.HashResource(sourceDef()), rawSourceList)
+			ruleMap["source"] = sourceSet
+		}
+		// destinations
+		excludeDestination := false
+		if rule.Destinations != nil && len(rule.Destinations.Destination) > 0 {
+			var destinationList []interface{}
+			if rule.Destinations.Excluded {
+				excludeDestination = true
+			}
+			for _, dest := range rule.Destinations.Destination {
+				destinationMap := map[string]interface{}{
+					"name":  dest.Name,
+					"type":  dest.Type,
+					"value": filterVdcId(dest.Value),
+				}
+				destinationList = append(destinationList, destinationMap)
+			}
+			destinationSet := schema.NewSet(schema.HashResource(destinationDef()), destinationList)
+			ruleMap["destination"] = destinationSet
+		}
+		// applications
+		if rule.Services != nil && len(rule.Services.Service) > 0 {
+			var applicationList []interface{}
+			for _, s := range rule.Services.Service {
+				applicationMap := map[string]interface{}{
+					"name":             s.Name,
+					"type":             s.Type,
+					"destination_port": stringOnNotNil(s.DestinationPort),
+					"source_port":      stringOnNotNil(s.SourcePort),
+					"protocol":         getDfwProtocolString(s.Protocol),
+					"value":            filterVdcId(s.Value),
+				}
+				applicationList = append(applicationList, applicationMap)
+			}
+			applicationSet := schema.NewSet(schema.HashResource(applicationDef()), applicationList)
+			ruleMap["application"] = applicationSet
+		}
+		// applied-to
+		if rule.AppliedToList != nil && len(rule.AppliedToList.AppliedTo) > 0 {
+			var appliedToList []interface{}
+			for _, a := range rule.AppliedToList.AppliedTo {
+				appliedToMap := map[string]interface{}{
+					"name":  a.Name,
+					"type":  a.Type,
+					"value": filterVdcId(a.Value),
+				}
+				appliedToList = append(appliedToList, appliedToMap)
+			}
+			appliedToSet := schema.NewSet(schema.HashResource(appliedToDef()), appliedToList)
+			ruleMap["applied_to"] = appliedToSet
+		}
+		if excludeDestination {
+			ruleMap["exclude_destination"] = true
+		}
+		if excludeSource {
+			ruleMap["exclude_source"] = true
+		}
+		rulesList = append(rulesList, ruleMap)
+	}
+
+	return d.Set("rule", rulesList)
+}
+
+// stringOnNotNil returns the contents of a string pointer
+// if the pointer is nil, returns an empty string
+func stringOnNotNil(p *string) string {
+	if p == nil {
+		return ""
+	}
+	return *p
+}
+
+func resourceVcdNsxvDistributedFirewallImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
```

Comment:
I see in other resources that we usually document how the Import works (example:` resourceVcdOrgVdcImport`), would it make sense to do the same here?

Reply:
done